### ISSUE TITLE
(v9) Badge: reconsidering aria-hidden true by default

### ISSUE THREAD

**Issue by tringakrasniqi** [microsoft/fluentui#21136]
### Environment Information

- **Package version(s)**:  v9 beta
- **Browser and OS versions**: all

### Describe the issue:

Badge applies `aria-hidden` to true in all cases by default even when it does provide meaningful information. A special case to meaningful info is the status on PresenceBadges, atm it is left up to the Avatar component to provide label for the badge as well which it might not be clear to users and result in issues. 

#### Expected behavior:
Badges that do have meaningful information or that contain icons should have proper labeling. Whereas, the others to be specified as presentation only.

It could be considered to add a prop which could be set if the badge is presentation only.

**Comment by tringakrasniqi** [comment]
# Research for Badge aria attributes

## Problem statement

The current implementation of Badge in FluentUI v9 applies `aria-hidden=true` by default. This causes for the badge to always be ignored by the screen readers, and/or the meaning of badge completely relies from other components (e.g Avatar). This can cause unwanted behaviours in some cases, for example:

![Badge "New" part of a heading](https://user-images.githubusercontent.com/12565826/151175263-6b4cd317-6360-42d5-87c6-c14b52e18105.png)

![Badges used as labels](https://user-images.githubusercontent.com/12565826/151175304-b7a89dd5-dc8e-413e-9738-10163c8f7872.png)


## Research

Researched a few component libraries that have `Badge` as a component, so to compare their implementation of this component in regards accessibility.

### Handling of Badge by Fabric and Northstar

Neither v8 or v0 have `Badge` as a component, but they are part of other components such as `Persona` and `Avatar`

#### [Northstar](https://fluentsite.z22.web.core.windows.net/0.54.0/components/avatar/definition)

For Northstar, PresenceBadges are part of the Avatar where the presence icon is shown if the `status` prop is provided.

This is DOM structure of the badge within an Avatar, in this case the wrapping span has `title="Available"` that is accessible to screen readers, but the icon itself is hidden:

```html
<span role="img" class="..." title="Available">
  <span class="...">
    <span role="img" aria-hidden="true" class="...">
      <svg role="presentation" focusable="false" class="..."></svg>
    </span>
  </span>
</span>
```

#### [Fabric](https://developer.microsoft.com/en-us/fluentui#/controls/web/persona)

The closest component to Badge is `Persona` which is a combination of Avatar and PresenceBadge of v9.

### Handling of Badge aria by other component libraries

The only component libraries that have some specification or special handling for aria are MUI and Lightening UI.

#### [MUI](https://mui.com/components/badges/#accessibility)

In MUI, the bagde is always as part of another component, e.g. IconButton that is responsible for giving a label.

```jsx
<IconButton aria-label={notificationsLabel(100)}>
  <Badge badgeContent={100} color="secondary">
    <MailIcon />
  </Badge>
</IconButton>
```

It is never intended to be used on its own, thus it is always hidden for screen readers. For cases where it is needed to stand on it own and present data, the `Chip` component is used.

#### [Lightening UI](https://www.lightningdesignsystem.com/components/badges/)

For Lightening, badges are meant to be used with short text that it can include icons. The text is exposed to screen readers and it is up to the developers to offer assistive text for the icons.

example:

```jsx
<span class="slds-badge slds-badge_inverse">
  <span class="...">
    <span
      class="slds-icon_container ..."
      title="Description of icon when needed"
    >
      <svg class="slds-icon ..." aria-hidden="true">
        ...
      </svg>
    </span>
  </span>
  423 Credits Available
</span>
```

Other component libraries that do not have any special handling for aria attributes are:

- [Atlaskit](https://atlassian.design/components/badge/examples)

  Example:

```html
<span class="css-92dv6g">5</span>
```

- [Ant UI](https://ant.design/components/badge/)
- [Bootstrap](https://getbootstrap.com/docs/4.3/components/badge/)
- [Evergreen](https://evergreen.segment.com/components/badges)
- [FAST](https://explore.fast.design/components/fast-badge)

## Ideas

1. Keeping the current behaviour with the `aria-hidden=true`, but updating the documentation where a more detaild information is provided for this. This will imply that application developers will be responsible for providing the correct behvaiour for assistive tools.

2. Adding logic which will determine if the badge is hidden based on the content it will have, for example:

- if the Badge has no content, meaning it is a dot: `<Badge/>`, then the `aria-hidden=true` is added
- if the Badge has content, it will be visible to screen readers: `<Badge>99+</Badge>`
- if the Badge has only an icon, then it is needed `aria-label` for the icon
- if the Badge has content and icon, then the icon has `aria-hidden=true` but the content is accessible to screen readers

This approach will need more exploring of different cases so that is will not cause unwanted behaviour.

3.  Adding a new prop to badge that it will determine if the Badge needs to be hidden or shown to the screen readers. Badges that contain important information, for example a number or icon will have the `hasContent` prop as true and it won't be hidden to screen readers. It is worth mentioning that badges that have icons will need `aria-label`.
    This prop should be a required one, so it is not missed by application developers.

```js
/**
 * If a badge has content that need to be accessible to screen readers then this should be set to true.
 * @defaultvalue false
 */
hasContent: boolean;
```


Any feedback or ideas will be highly appreciated 🙏
